src/bitcoin-cli.cpp

```diff
@@ -286,7 +286,30 @@ int CommandLineRPC(int argc, char *argv[])
         if (args.size() < 1)
             throw std::runtime_error("too few parameters (need at least command)");
         std::string strMethod = args[0];
-        UniValue params = RPCConvertValues(strMethod, std::vector<std::string>(args.begin()+1, args.end()));
+        args.erase(args.begin()); // Remove trailing method name from arguments vector
+
+        // Check for named arguments.
+        // Note that ParseParameters stops at the first argument that doesn't
+        // start with a switch character, which happens to be the method. This
+        // starts scanning after the method, so arguments before and after the
+        // command will never interfere.
+        bool haveNamed = false;
+        bool havePositional = false;
+        for (const std::string &s: args) {
+            if (s.size() >= 1 && IsSwitchChar(s[0]))
```

Comment:
I believe this breaks passing in negative numbers as a positional argument. For example, the following cli command is valid prior to this commit:

`bitcoin-cli getnetworkhashps -1`

and returns the estimated network hashes per second in the current difficulty window. After this commit, the method fails:

```
vagrant@vagrant-ubuntu-trusty-64:~$ bitcoin-cli getnetworkhashps -1
error: No '=' in named argument, a value needs to be provided for every argument
```

I haven't checked whether there are arguments that accept negative numbers in other RPC methods.